iquip/apps/builder.py

```diff
@@ -365,22 +355,146 @@ def get_state(self, argInfo: Dict[str, Any]) -> Dict[str, Any]:
+        randomize: QCheckBox for randomize arguments inside state.
+        layout: The layout of _RangeScan widget.
+    """
+    def __init__(
+        self,
+        procdesc: Dict[str, Any],
+        state: Dict[str, Any],
+        parent: Optional[QWidget] = None
+        ): # pylint: disable=too-many-statements
+        """Extended.
+
+        Args:
+            procdesc: Each key and its value are as follows.
+              unit: The unit of the number value.
+              scale: The scale factor that is multiplied to the number value.
+              global_step: The step between values changed by the up and down button.
+              global_min: The minimum value. (default=float("-inf"))
+              global_max: The maximum value. (default=float("inf"))
+                If min > max, then they are swapped.
+              ndecimals: The number of displayed decimals.
+            state: Each key and its value as follows.
+              start: The start point for the RangeScan sequence.
+              stop: The end point for the RangeScan sequence.
+              npoints: The number of points in the RangeScan squence.
+              randomize: A boolean value that decides whether to shuffle the RangeScan sequence.
+        """
+        super().__init__(parent=parent)
+        self.scale = procdesc["scale"]
+        self.state = state
+        self.layout = QGridLayout(self)
+        self.start = QDoubleSpinBox(self)
+        self.apply_properties(self.start, procdesc)
+        self.start.setValue(state["start"] / self.scale)
+        self.npoints = QSpinBox(self)
+        self.npoints.setMinimum(1)
+        self.npoints.setMaximum((1 << 31) - 1)
+        self.npoints.setValue(state["npoints"])
+        self.stop = QDoubleSpinBox(self)
+        self.apply_properties(self.stop, procdesc)
+        self.stop.setValue(state["stop"] / self.scale)
+        self.randomize = QCheckBox("Randomize", self)
+        self.randomize.setChecked(state["randomize"])
+        #layout
+        self.layout.addWidget(QLabel("start:", self), 0, 0)
+        self.layout.addWidget(self.start, 0, 1)
+        self.layout.addWidget(QLabel("npoints:", self), 1, 0)
+        self.layout.addWidget(self.npoints, 1, 1)
+        self.layout.addWidget(QLabel("stop:", self), 2, 0)
+        self.layout.addWidget(self.stop, 2, 1)
+        self.layout.addWidget(self.randomize, 3, 1)
+        self.start.valueChanged.connect(self.update_start)
+        self.npoints.valueChanged.connect(self.update_npoints)
+        self.stop.valueChanged.connect(self.update_stop)
+        self.randomize.stateChanged.connect(self.update_randomize)
+
+    def apply_properties(self, widget: QDoubleSpinBox, procdesc: Dict[str, Any]):
+        """Adds properties to the SpinBox widget.
+
+        Attributes:
+            widget: A QDOubleSpinWidget that has properties to set. 
+        """
+        widget.setDecimals(procdesc["ndecimals"])
+        if procdesc["global_min"] is not None:
```

Comment:
Accessing each value using `procdesc[...]` looks messy.
How about implementing like `_NumberEntry`?

Reply:
I'd like to unify the implementation way in `_NumberEntry` and this method, if there is no more efficient way.

Reply:
That seems good.